File: shopcar/src/main/java/com/cxyz/car/presenter/StorePresenter.java
package com.cxyz.car.presenter;

import com.cxyz.car.data.domain.CommentsItem;
import com.cxyz.car.data.model.IStoreModel;
import com.cxyz.car.data.model.impl.StoreModelImpl;
import com.cxyz.car.presenter.view.IStoreView;
import com.cxyz.mvp.ipresenter.IBasePresenter;

import java.util.List;

public class StorePresenter extends IBasePresenter<IStoreModel, IStoreView> {
    @Override
    public IStoreModel createModel() {
        return new StoreModelImpl();
    }

    public void fecth(){
        mIModle.loadData(new IStoreModel.OnLoadListener() {
            @Override
            public void complete(List<CommentsItem> commentsItemList) {
                mIView.showStoreComments(commentsItemList);
            }
        });
    }
}

File: main/src/main/java/com/cxyz/main/presenter/StorePresenter.kt
package com.cxyz.main.presenter

import com.cxyz.main.data.model.IStoreModel
import com.cxyz.main.data.model.impl.StoreModelImpl
import com.cxyz.main.presenter.view.IStoreView
import com.cxyz.mvp.ipresenter.IBasePresenter

class StorePresenter : IBasePresenter<IStoreModel,IStoreView>() {
    override fun createModel(): IStoreModel {
        return StoreModelImpl()
    }

    /**
     * 根据id获取商铺的信息
     */
    fun getStoreInfo (id: String) {

    }
}